File: src/1147-Longest-Chunked-Palindrome-Decomposition/1147.py
class Solution:
    def longestDecomposition(self, text: str) -> int:
        n = len(text)
        for i in range(1, n//2 + 1):
            if text[:i] == text[n-i:]:
                return 2 + self.longestDecomposition(text[i:n-i])
        return 0 if n == 0 else 1 

File: src/1147-Longest-Chunked-Palindrome-Decomposition/1147.go
func longestDecomposition(text string) int {
    var n = len(text);
    for i := 1; i <= n/2; i++ {
        if text[:i] == text[n-i:] {
            return 2 + longestDecomposition(text[i:n-i])
        }
    }
    if n == 0 {
        return 0
    }
    return 1
}